tests/test_model_manager.py

```diff
@@ -233,40 +367,304 @@ def test_manual_parameters(self):
+            input [
+            {
+                name: "INPUT__0"
+                data_type: TYPE_FP32
+                dims: [16]
+            }
+            ]
+            output [
+            {
+                name: "OUTPUT__0"
+                data_type: TYPE_FP32
+                dims: [16]
+            }
+            ]
+            instance_group [
+            {
+                kind: KIND_CPU
+                count: 4
+            }
+            ]
+            """
+
+        expected_ranges = [{
+            'instances': [1],
+            'kind': ["KIND_GPU"],
+            'batching': [None, 0, 1, 2],
+            'batch_sizes': [1],
+            'max_batch_size': [8],
+            'concurrency': [1, 2, 4]
+        }, {
+            'instances': [4],
+            'kind': ["KIND_CPU"],
+            'batching': [None],
+            'batch_sizes': [1],
+            'max_batch_size': [8],
+            'concurrency': [1, 2, 4]
+        }]
 
+        yaml_content = """
+            run_config_search_max_concurrency: 4
+            run_config_search_max_preferred_batch_size: 2
+            run_config_search_max_instance_count: 1
+            run_config_search_preferred_batch_size_disable : False
+            run_config_search_disable: False
+            profile_models: test_model
+            """
+        self._test_model_manager(yaml_content, expected_ranges)
 
-    def _test_model_manager(self, yaml_content, expected_values):
+    def _test_model_manager(self, yaml_content, expected_ranges):
         """ 
         Test helper function that passes the given yaml_content into
-        model_manager and asserts that the number of run_configs generated
-        is equal to the passed in expected_count 
+        model_manager, runs the model, and confirms the result is as expected
+        based on a full cartesian product of the lists in the input list of 
+        dicts expected_ranges
         """
 
-        args = [
-            'model-analyzer', 'profile', '--model-repository', 'cli_repository',
-            '-f', 'path-to-config-file', '--profile-models', 'vgg11'
-        ]
+        # Use mock model config or else TritonModelAnalyzerException will be thrown as it tries to read from disk
+        self.mock_model_config = MockModelConfig(self._model_config_protobuf)
+        self.mock_model_config.start()
+        config = self._evaluate_config(self._args, yaml_content)
 
-        config = self._evaluate_config(args, yaml_content)
         state_manager = AnalyzerStateManager(config, MagicMock())
-        model_manager = ModelManagerSubclass(config, MagicMock(), MagicMock(), MagicMock(), MagicMock(), state_manager)
+        model_manager = ModelManagerSubclass(config, MagicMock(), MagicMock(),
+                                             MagicMock(), MagicMock(),
+                                             state_manager)
 
         model_manager.run_model(config.profile_models[0])
+        self.mock_model_config.stop()
```

Comment:
Why did this get take out of the tearDown function?

Reply:
Before it was a fixed protobuf, so start and stop could be in setup and teardown. I added the ability to supply a different/custom protobuf, so I had to move the start/stop into _test_model_manager()